src/components/TodoModal/TodoModal.tsx

```diff
@@ -1,12 +1,45 @@
-import React from 'react';
+import React, { useEffect, useState } from 'react';
 import { Loader } from '../Loader';
+import { getUser } from '../../api';
+import { User } from '../../types/User';
+import { Todo } from '../../types/Todo';
+
+interface TodoModalProps {
+  handleCloseModal: () => void;
+  selectedUserId: number;
+  selectedTodo: Todo | null;
+}
+export const TodoModal: React.FC<TodoModalProps> = ({
+  handleCloseModal,
+  selectedUserId,
+  selectedTodo,
+}) => {
+  const [isLoading, setIsLoading] = useState(true);
+  const [user, setUser] = useState<User | null>(null);
+
+  async function handleLoadUserInfo(userId: number) {
+    const loadedUser = await getUser(userId);
+
+    if (!loadedUser) {
+      handleCloseModal();
+
+      return;
+    }
+
+    setUser(loadedUser);
+    setIsLoading(false);
```

Comment:
Ensure `setIsLoading(false)` is called in both the success and failure paths of `handleLoadUserInfo` to correctly update the loading state.